client/my-sites/themes/theme-showcase.jsx

```diff
@@ -207,7 +212,10 @@ class ThemeShowcase extends React.Component {
 		let callback = () => null;
 		// In this state: tabFilter = [ Recommended | ##All(1)## ]  tier = [ All(2) | Free | ##Premium## ]
 		// Clicking "Recommended" forces tier to be "all", since Recommend themes cannot filter on tier.
-		if ( 'recommended' === tabFilter && 'all' !== this.props.tier ) {
+		if (
+			'recommended' === tabFilter ||
+			( 'uploaded' === tabFilter && 'all' !== this.props.tier )
+		) {
```

Comment:
I think the check should be like this:
```js
		if (
			( 'recommended' === tabFilter || 'uploaded' === tabFilter ) &&
			'all' !== this.props.tier
		) {
```
That is, if the tier is not all, and we change to either "Recommended" or "Uploaded", we force the tier to be all.

Although generally, it would be preferable if adding a tab meant that you weren't required to add it to all of these checks, but that can come in another PR..